### PR TITLE
DOI 759 ev dashboard packages npm publish

### PR DESCRIPTION
**Pull request by royki**
- Added GitHub CI
- Package names changed as below
```
@ev-dashboard-client/asset-registrar -> @energyweb/ev-registry
@ev-dashboard-client/did-hydrator -> @energyweb/ev-did-hydrator
@ev-dashboard-client/key-manager -> @energyweb/ev-key-manager
@ev-dashboard-client/prequalification-client -> @energyweb/ev-prequalification
@ev-dashboard-client/signer-provider-interface -> @energyweb/ev-signer-interface
@ev-dashboard-client/asset-operator-server -> @energyweb/ev-operator-server
@ev-dashboard-client/asset-operator-cli -> @energyweb/ev-operator-cli
@ev-dashboard-client/contract-deployer -> @energyweb/ev-contract-deployer [this one will not be published]
```
- Packages published through CI
https://www.npmjs.com/package/@energyweb/ev-asset-operator-server
https://www.npmjs.com/package/@energyweb/ev-asset-operator-cli
https://www.npmjs.com/package/@energyweb/ev-registry
https://www.npmjs.com/package/@energyweb/ev-did-hydrator
https://www.npmjs.com/package/@energyweb/ev-key-manager
https://www.npmjs.com/package/@energyweb/ev-prequalification
https://www.npmjs.com/package/@energyweb/ev-signer-interface
- version policy not applied (@jrhender can you please take a look)

- `publish` branch needs to be created to publish npm packages via CI.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    needs: cancel-previous
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: '14'
+          registry-url: https://registry.npmjs.org/
+
```
Comment (jrhender) [inline .github/workflows/publish.yml:31]: Could we add a call to rush change -v in here to make sure that the change files have been generated?
```suggestion
- name: Rush change verification
        run: node common/scripts/install-run-rush.js change -v
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+on:
+  push:
+    branches:
+      - 'DOI-759-ev-dashboard-client-npm-publish'
+  pull_request:
+    branches:
+      - '*'
+jobs:
+  cancel-previous:
```
Comment (jrhender) [inline .github/workflows/publish.yml:11]: Just curious, what does this job do? It lets you cancel a package publish while it is happening?

**Review by jrhender — COMMENTED** [review]
Looks really good! A few questions/suggestions

File: .github/workflows/publish.yml (hunk 1 above)
Comment (jrhender) [reply .github/workflows/publish.yml:31]: @royki Added this here: https://github.com/energywebfoundation/ev-dashboard-client/commit/65fee32b4d53985e9479c86e660bd6ad86f610cd

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - 'DOI-759-ev-dashboard-client-npm-publish'
+  pull_request:
+    branches:
+      - '*'
```
Comment (jrhender) [inline .github/workflows/publish.yml:9]: @royki  Does this mean that the publish will run on any pull_request on any branch?

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - 'DOI-759-ev-dashboard-client-npm-publish'
```
[the remark this answers is not in the document]
Comment (royki) [reply .github/workflows/publish.yml:6]: Yes, we can change it to publish or may be create a new branch call `publish` and push on `publish` branch will publish to npm registry.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (royki) [reply .github/workflows/publish.yml:11]: This action cancels previous runs for one or more branches/prs associated with a workflow, effectively limiting the resource consumption of the workflow to one per branch.
https://github.com/marketplace/actions/cancel-previous-workflow-runs

File: .github/workflows/publish.yml (hunk 3 above)
Comment (royki) [reply .github/workflows/publish.yml:9]: We can remove this. This doesn't need actually.

**Pull request merged**